client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -0,0 +1,389 @@
+package ai.verta.repository
+
+import ai.verta.client.{getPersonalWorkspace, urlEncode}
+import ai.verta.blobs._
+import ai.verta.swagger.client.ClientSet
+import ai.verta.swagger._public.modeldb.versioning.model._
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success, Try}
+import scala.collection.mutable.HashMap
+
+/** Commit within a ModelDB Repository
+ *  There should not be a need to instantiate this class directly; please use Repository.getCommit methods
+ */
+class Commit(
+  val clientSet: ClientSet, val repo: VersioningRepository,
+  var commit: VersioningCommit, var commit_branch: Option[String] = None
+) {
+  var saved = true // whether the commit instance is saved to database
+  var loaded_from_remote = false // whether blob has been retrieved from remote
+  var blobs = new HashMap[String, VersioningBlob]() // mutable map for storing blobs
+
+  /** Retrieve commit's blobs from remote
+   */
+  private def load_blobs()(implicit ec: ExecutionContext): Unit = {
+    if (!loaded_from_remote) {
+      val ids: List[String] = commit.commit_sha match {
+        case Some(v) => List(v)
+        case None => commit.parent_shas.get
```

Comment:
This seems wrong to me. @convoliution do we load all parents?

Reply:
Yes; if the `Commit` doesn't have its own ID yet, the Python Client [iterates through the parents and loads in all of their blobs](https://github.com/VertaAI/modeldb/blob/master/client/verta/verta/_repository/commit.py#L62).

But note that this is (or should be) only the immediate parents, not the entire branch history.